extensions/http-curl/client/HTTPClient.h

```diff
@@ -172,6 +172,20 @@ class HTTPClient : public BaseHTTPClient, public core::Connectable {
 
   void setPostSize(size_t size);
 
+  void setHTTPProxy(const utils::HTTPProxy &proxy) override {
+    if (!proxy.host.empty()) {
+      curl_easy_setopt(http_session_, CURLOPT_PROXY, proxy.host.c_str());
+      curl_easy_setopt(http_session_, CURLOPT_PROXYPORT, proxy.port);
+      if (!proxy.username.empty()) {
+        /* allow whatever auth the proxy speaks */
```

Comment:
Might help to change the comments since they're identical to https://curl.haxx.se/libcurl/c/CURLOPT_PROXYAUTH.html